frontend/context/StakingProgramProvider.tsx

```diff
@@ -1,91 +1,79 @@
-/**
- * This context provider is responsible for determining the current active staking program, if any.
- * It does so by checking if the current service is staked, and if so, which staking program it is staked in.
- * It also provides a method to update the active staking program id in state.
- */
-
 import { useQuery, useQueryClient } from '@tanstack/react-query';
 import { createContext, PropsWithChildren, useCallback } from 'react';
 
-import { AGENT_CONFIG } from '@/config/agents';
-import { GNOSIS_CHAIN_CONFIG } from '@/config/chains';
 import { FIVE_SECONDS_INTERVAL } from '@/constants/intervals';
 import { REACT_QUERY_KEYS } from '@/constants/react-query-keys';
 import { StakingProgramId } from '@/enums/StakingProgram';
-import { useService } from '@/hooks/useService';
-import { useServices } from '@/hooks/useServices';
+import { useAgent } from '@/hooks/useAgent';
+import { useChainId } from '@/hooks/useChainId';
+import { useServiceId } from '@/hooks/useService';
 import { Nullable } from '@/types/Util';
 
 export const INITIAL_DEFAULT_STAKING_PROGRAM_ID = StakingProgramId.PearlBeta;
 
 export const StakingProgramContext = createContext<{
-  activeStakingProgramId?: StakingProgramId | null;
-  defaultStakingProgramId: StakingProgramId;
+  isActiveStakingProgramLoaded: boolean;
+  activeStakingProgramId?: Nullable<StakingProgramId>;
 }>({
-  activeStakingProgramId: undefined,
-  defaultStakingProgramId: INITIAL_DEFAULT_STAKING_PROGRAM_ID,
+  isActiveStakingProgramLoaded: false,
+  activeStakingProgramId: null,
 });
```

Comment:
think will also need to be refactored, assumes only one active staking program, where there will be multiple services